subsys/fs/shell.c

```diff
@@ -334,6 +334,66 @@ static int cmd_read(const struct shell *shell, size_t argc, char **argv)
 	return 0;
 }
 
+static int cmd_cat(const struct shell *shell, size_t argc, char **argv)
+{
+	char path[MAX_PATH_LEN];
+	struct fs_dirent dirent;
+	struct fs_file_t file;
+	int count;
+	int err;
+
+	for (size_t i = 1; i < argc; ++i) {
+		create_abs_path(argv[i], path, sizeof(path));
+
+		err = fs_stat(path, &dirent);
+		if (err) {
+			shell_error(shell, "Failed to obtain file %s (err: %d)",
+					path, err);
+			continue;
+		}
+
+		if (dirent.type != FS_DIR_ENTRY_FILE) {
+			shell_error(shell, "Note a file %s", path);
+			continue;
+		}
+
+		fs_file_t_init(&file);
+		err = fs_open(&file, path, FS_O_READ);
+		if (err) {
+			shell_error(shell, "Failed to open %s (%d)", path, err);
+			continue;
+		}
+
+		count = INT_MAX;
+
+		while (count > 0) {
```

Comment:
The `count` does not seem usable in any way. You do not set it as a command parameter to limit number of bytes/lines that would go out. It is used in `fs_read` call in line 373 but all it does it to limit output of file to 2^(sizeof(inf)-1) bytes.
Either make command line option or leave it out; you can check if loop should run by checking the `fs_read` still returning bytes..